src/Abstractions/NexusMods.Abstractions.Loadouts/CollectionGroup.cs

```diff
@@ -0,0 +1,42 @@
+using DynamicData.Kernel;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+
+namespace NexusMods.Abstractions.Loadouts;
+
+/// <summary>
+/// A definition of a collection. This could be a user's collection, a Nexus Mods collection or perhaps a collection from some other source.
+/// </summary>
+[Include<LoadoutItemGroup>]
+public partial class CollectionGroup : IModelDefinition
+{
+    private const string Namespace = "NexusMods.Abstractions.Loadouts.CollectionGroup";
+    
+    /// <summary>
+    /// Mostly a marker attribute to find all the collections on a given loadout. Likely faster than searching for all the loadout items
+    /// and filtering them by their attributes
+    /// </summary>
+    public static readonly ReferenceAttribute<Loadout> Loadout = new(Namespace, nameof(Loadout));
+
+    /// <summary>
+    /// In this UI this is reflected as "My Collection", it's a group of all mods that the user has installed outside of any
+    /// other collection.
+    /// </summary>
+    public static readonly MarkerAttribute UserCollection = new(Namespace, nameof(UserCollection));
```

Comment:
The semantics of this should be different. A collection is either read-only, because it was downloaded, or mutable, because the user created it. We can have more than 1 "user collection", the "My Collection" is just the default one we create.

There isn't anything special about the "My Collection", the only code that actually cares about that is the code that creates the Loadout. Everything else will either be passed a collection to use or use the first mutable collection as a default.

The comment here should be updated to reflect these semantics. Currently it suggests that there is only one user collection, when the user can actually create as many as they want.